kube-core/src/schema/transform_one_of.rs

```diff
@@ -0,0 +1,181 @@
+use std::ops::DerefMut;
+
+use schemars::transform::Transform;
+
+use crate::schema::{Schema, SchemaObject, SubschemaValidation};
+
+/// Merge oneOf subschema enums and consts into a schema level enum.
+///
+/// Used for correcting the schema for tagged enums with unit variants.
+///
+/// NOTE: Subschema descriptions are lost when they are combined into a single
+/// enum of the same type.
+///
+/// This will return early without modifications unless:
+/// - There are `oneOf` subschemas (not empty).
+/// - Each subschema contains an `enum` or `const`.
```

Comment:
Possibly this can be reduced in scope if we use the existing [ReplaceConstValue](https://docs.rs/schemars/latest/schemars/transform/struct.ReplaceConstValue.html)? it [doesn't do a lot](https://docs.rs/schemars/latest/src/schemars/transform.rs.html#369) though.